tests/drivers/sensor/generic/src/main.c

```diff
@@ -111,11 +111,22 @@ void test_sensor_get_channels(void)
 	dev = device_get_binding(DUMMY_SENSOR_NAME);
 	zassert_not_null(dev, "failed: dev is null.");
 
+	zassert_equal(sensor_sample_fetch_chan(dev, chan_elements[0].chan),
+				RETURN_SUCCESS,	"fail to fetch sample.");
+
+	zassert_equal(sensor_channel_get(dev, chan_elements[0].chan,
+				&data), RETURN_SUCCESS, "fail to get channel.");
+
+	zassert_equal(data.val1, chan_elements[0].data.val1,
+				"the data is not match.");
+	zassert_equal(data.val2, chan_elements[0].data.val2,
+				"the data is not match.");
+
 	zassert_equal(sensor_sample_fetch(dev), RETURN_SUCCESS,
 			"fail to fetch sample.");
 
 	/* Get and check channels value. */
-	for (int i = 0; i < TOTAL_CHAN_ELEMENTS; i++) {
+	for (int i = 1; i < TOTAL_CHAN_ELEMENTS; i++) {
```

Comment:
We don't need to run testing for channel 0 any more?

Reply:
Sensor data in chanel 0 is taken into ram by the supplement api sensor_sample_fetch_chan() and obtained by cpu use sensor_channel_get() at the begin of this testcase. 
So I think the date in chanel 0 is no need to be get by cpu twice.